sw/device/lib/dif/dif_entropy.h

```diff
@@ -0,0 +1,678 @@
+ * @param snapshot A snapshot to restore from.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_irq_restore_all(
+    const dif_entropy_t *entropy, const dif_entropy_irq_snapshot_t *snapshot);
+
+/**
+ * Forces a particular alert, causing it to be emitted as if the hardware had
+ * done so.
+ *
+ * @param entropy An entropy source handle.
+ * @param alert An alert type.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_alert_force(const dif_entropy_t *entropy,
+                                             dif_entropy_alert_t alert);
+
+/**
+ * Performs an override read from the entropy pipeline.
+ *
+ * This function pauses entropy flow out of the pre-conditioner FIFO and
+ * instead flows words into `buf`. Normal operation of the entropy pipeline
+ * will not resume until `dif_entropy_fifo_reconnect()` is called.
+ *
+ * `buf` may be `NULL`; in this case, reads will be discarded.
+ *
+ * @param entropy An entropy source handle.
+ * @param[out] buf A buffer to fill with words from the pipeline.
+ * @param len The number of words to read into `buf`.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_fifo_read(const dif_entropy_t *entropy,
+                                           uint32_t *buf, size_t len);
+
+/**
+ * Performs an override write to the entropy pipeline.
+ *
+ * This function pauses entropy flow into the pre-conditioner FIFO and
+ * instead flows words out of `buf`. Normal operation of the entropy pipeline
+ * will not resume until `dif_entropy_override_reset()` is called.
+ *
+ * @param entropy An entropy source handle.
+ * @param buf A buffer to push words from into the pipeline.
+ * @param len The number of words to read from `buf`.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_fifo_write(const dif_entropy_t *entropy,
+                                            const uint32_t *buf, size_t len);
+
+/**
+ * Gets the current number of entries in the pre-conditioner FIFO.
+ *
+ * This function pauses the flow through the FIFO.
+ *
+ * @param entropy An entropy source handle.
+ * @param[out] len The number of words in the FIFO.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_fifo_get_len(const dif_entropy_t *entropy,
+                                              uint8_t *len);
+
+/**
+ * Gets the current capacity of the pre-conditioner FIFO.
+ *
+ * @param entropy An entropy source handle.
+ * @param[out] capacity The number of words of capacity in the FIFO.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_fifo_get_capacity(const dif_entropy_t *entropy,
+                                                   uint8_t *capacity);
+
+/**
+ * Sets the current capacity of the pre-conditioner FIFO.
+ *
```

Comment:
nit: Can you mention that (or better define a constant for) the max value: `0x40`.